src/nvim/highlight_group.c

```diff
@@ -301,6 +301,11 @@ static const char *highlight_init_both[] = {
   "default link @tag         Tag",
   "default link @tag.builtin Special",
 
+  // Higlight "===" and "---" heading delimiters specially.
+  // This is for :help but also presumably useful for markdown etc.
+  "default @markup.heading.1.delimiter guibg=bg guifg=bg guisp=fg gui=underdouble,nocombine ctermbg=NONE ctermfg=NONE cterm=underdouble,nocombine",
```

Comment:
One issue I see with this is that this will not play nicely with other colorschemes since the `guisp` is hard-coded (and likely different from the `@markup.heading.1` group).

(And since it's a "secret" highlight group, colorschemes will not override it.)

Reply:
> since the `guisp` is hard-coded

how is it any more "hard coded" than any other `default` highlight attribute?